i18n/create_scratch_msgs.js

```diff
@@ -0,0 +1,66 @@
+const fs = require('fs');
+const path = require('path');
+const glob = require('glob');
+const assert = require('assert');
+
+// Globals
+const PATH_INPUT = path.resolve(__dirname, '../msg/json/*.json');
+const PATH_OUTPUT = path.resolve(__dirname, '../msg');
+
+let en = fs.readFileSync(path.resolve(__dirname, '../msg/json/en.json'));
+en = JSON.parse(en);
+const enKeys = Object.keys(en).sort().toString();
+
+// Check that translation is valid:
+// entry: array [key, translation]  corresponding to a single string from <locale>.json
+// - messages with placeholders have the same number of placeholders
+// - messages must not have newlines embedded
+const validateEntry = function (entry) {
+    const re = /(%\d)/g;
+    const [key, translation] = entry;
+    const placeholdersCount = en[key].match(re) ? en[key].match(re).length : 0;
+    if (placeholdersCount > 0) {
+      const tMatch = translation.match(re);
+      const enMatch = en[key].match(re);
+      assert.notStrictEqual(tMatch, null, `${key} is missing a placeholder: ${translation}`);
+      assert.strictEqual(
+          tMatch.sort().toString(),
+          enMatch.sort().toString(),
+          `${key} is missing or has duplicate placeholders: ${translation}`
+      );
+    }
+    assert.strictEqual(translation.match(/[\n]/), null, `${key} contains a newline character ${translation}`);
+};
+
+const validate = function (json, name) {
+    assert.strictEqual(Object.keys(json).sort().toString(), enKeys, `${name}: Locale json keys do not match en.json`);
+    Object.entries(json).forEach(validateEntry);
+};
+
+let file = `// This file was automatically generated.  Do not modify.
+
+'use strict\';
```

Comment:
Should this first `'` also be escaped? Alternatively, should none of these be escaped since we're using the `` ` `` formatted strings?

Reply:
The escape is not necessary inside a template literal (but doesn't hurt). It should be consistent, though.